packages/dev/core/src/Misc/uvprojector.ts

```diff
@@ -0,0 +1,136 @@
+
+        const boundingBox = target.getBoundingInfo().boundingBox
+        const _width = (boundingBox.maximum.x - boundingBox.minimum.x) * scalefact;
+        const _height = (boundingBox.maximum.y - boundingBox.minimum.y) * scalefact;
+        const _depth = (boundingBox.maximum.z - boundingBox.minimum.z) * scalefact;
+
+        if(!reference){
+            reference = CreateBox("box-unwrap", {width: _width, height: _height, depth: _depth})
+            reference.position.copyFrom(boundingBox.center);
+            reference.bakeCurrentTransformIntoVertices();
+            reference.visibility = 0.3;
+        }else{
+            reference.scaling.set(_width, _height, _depth);
+            reference.position.copyFrom(boundingBox.center);
+            reference.bakeCurrentTransformIntoVertices();
+            reference.visibility = 0.3;
+        }
+
+        function getFaceNormal(pt1: any, pt2: any, pt3: any): any{ 
+            let convertToArray = false;
+            if(pt1 instanceof Array || pt1 instanceof Float32Array){
+                pt1 = new Vector3().fromArray(pt1, 0);
+                pt2 = new Vector3().fromArray(pt2, 0);
+                pt3 = new Vector3().fromArray(pt3, 0);
+                convertToArray = true;
+            }    
+            const normal = Vector3.Zero();
+            Vector3.CrossToRef(pt2.subtract(pt1), pt3.subtract(pt1), normal)
+            normal.normalize();
+
+            if(convertToArray){
+                return normal.asArray();
+            }else{
+                return normal;
+            }
+        }
+
+
+        function predicate(mesh: AbstractMesh): boolean{
+            if (mesh == reference){
+                return true;
+            }
+            return false;
+        }
+        
+        //use ray to project the uv data from the reference mesh to the target mesh
+        for(let i = 0; i < indices.length; i++){
+    
+            if(i % 3 == 0){
+                let rayDir = new Vector3(0, 1, 0);
+    
+                let faceDir = getFaceNormal(vec[indices[i]], vec[indices[i + 1]], vec[indices[i + 2]]);
+                faceDir.normalize();
+                faceDir = new Vector3(Math.abs(faceDir.x), Math.abs(faceDir.y), Math.abs(faceDir.z))
+                
+                if(faceDir.x > faceDir.y && faceDir.x > faceDir.z){
+                    rayDir = new Vector3(1, 0, 0);
+                }else if(faceDir.y > faceDir.x && faceDir.y > faceDir.z){
+                    rayDir = new Vector3(0, 1, 0);
+                }else if(faceDir.z > faceDir.x && faceDir.z > faceDir.y){
+                    rayDir = new Vector3(0, 0, 1);
+                }
+    
+                rayDir = rayDir.normalize();
+                const rayLength = rayDir.length() * Math.abs(_width * _height * _depth);
+    
+                const faceVerts = [vec[indices[i]], vec[indices[i + 1]], vec[indices[i + 2]]];
+                const trigInd = [indices[i], indices[i + 1], indices[i + 2]]

```

Comment:
Don't create this temporary array and simply use `indices[i + n]` in the loop below, to avoid GC pressure.